test/pool/test_max_pool2d_fwd.cpp

```diff
@@ -49,15 +49,20 @@ class TestMaxPool2dFwd : public ::testing::Test
 using true_t  = std::integral_constant<bool, true>;
 using false_t = std::integral_constant<bool, false>;
```

Comment:
Why we need this ` std::integral_constant<bool, Cond>;` Can we use `false\true` directly in `std::conditional_t` ?

Reply:
Because this gtest class needs the true/false values passed as a type and not as a value, so I had to wrap it into `std::integral_constant` to make it compile